pkg/storage/mysql/mysqlstore.go

```diff
@@ -0,0 +1,275 @@
+	_, err = tx.Exec(createTableStmt)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create table %s: %w", tableName, err)
+	}
+
+	store := &sqlDBStore{
+		db:        p.db,
+		tableName: tableName}
+
+	p.dbs[name] = store
+
+	return store, nil
+}
+
+// Close closes the provider.
+func (p *Provider) Close() error {
+	p.Lock()
+	defer p.Unlock()
+
+	for _, store := range p.dbs {
+		err := store.db.Close()
+		if err != nil {
+			return fmt.Errorf(failToCloseProviderErrMsg+": %w", err)
+		}
+	}
+
+	if err := p.db.Close(); err != nil {
+		return err
+	}
+
+	p.dbs = make(map[string]*sqlDBStore)
+
+	return nil
+}
+
+// CloseStore closes a previously opened store
+func (p *Provider) CloseStore(name string) error {
+	p.Lock()
+	defer p.Unlock()
+
+	store, exists := p.dbs[name]
+	if !exists {
+		return nil
+	}
+
+	delete(p.dbs, name)
+
+	return store.db.Close()
+}
+
+// Put stores the key and the record
+func (s *sqlDBStore) Put(k string, v []byte) error {
+	if k == "" {
+		return errKeyRequired
+	}
+
+	//nolint: gosec create upsert query to insert the record, checking whether the key is already mapped to a value in the store.
+	createStmt := "INSERT INTO " + s.tableName + " VALUES (?, ?) ON DUPLICATE KEY UPDATE value=?"
+	// executing the prepared insert statement
+	_, err := s.db.Exec(createStmt, k, v, v)
+	if err != nil {
+		return fmt.Errorf("failed to insert key and value record into %s %w ", s.tableName, err)
+	}
+
+	return nil
+}
+
+// Get fetches the record based on key
+func (s *sqlDBStore) Get(k string) ([]byte, error) {
+	if k == "" {
+		return nil, errKeyRequired
+	}
+
+	var res result
+	// nolint: gosec select query to fetch the record by key
+	err := s.db.QueryRow("SELECT * FROM "+s.tableName+" "+
+		" WHERE `key` = ?", k).Scan(&res.key, &res.value)
+	if err != nil {
+		if strings.Contains(err.Error(), sqlDBNotFound) {
+			return nil, storage.ErrDataNotFound
+		}
+
+		return nil, err
+	}
+
+	return res.value, nil
+}
+
+// Delete will delete record with k key
+func (s *sqlDBStore) Delete(k string) error {
+	if k == "" {
+		return errKeyRequired
+	}
+	// nolint: gosec delete query to delete the record by key
+	_, err := s.db.Exec("DELETE FROM "+s.tableName+" WHERE `key`= ?", k)
+
+	if err != nil {
+		return fmt.Errorf("failed to delete row %w", err)
+	}
+
+	return nil
+}
+
+type sqlDBResultsIterator struct {
+	store      *sqlDBStore
+	resultRows *sql.Rows
+	err        error
+}
+
+func (s *sqlDBStore) Iterator(startKey, endKey string) storage.StoreIterator {
+	// sub query to fetch the all the keys that have start and end key reference, simulating range behavior.
+	queryStmt := "SELECT DISTINCT `key`, value FROM " +
+		"(SELECT * FROM " + s.tableName + " WHERE `key` LIKE ? " +
+		"UNION ALL SELECT * FROM " + s.tableName + " WHERE `key` LIKE ?) as d"
+
+	if startKey != "" && endKey != "" {
+		return s.prepareQueryResult(queryStmt, "%"+startKey+"%", "%"+endKey+"%")
+	}
+
+	return s.prepareQueryResult(queryStmt, startKey, endKey)
+}
```

Comment:
1. replace `"%"+startKey+"%"` with `startKey+"%"` 
2.  replace `"%"+endKey+"%` with `"endKey+"%` 
3. endKey has suffix `!!`. Take a look of how we are using it `storage.EndKeySuffix`.

 The query will work. You are using LIKE again. 
The payload can be like this:
| key | val |
| ---- | ----|
| key_1 | val_1|
| key_2 | val_2|
| opps | val_3|
| key_4 | val_4|
| key_5 | val_5|
Make sure that `Iterator("key_2","key_4!!")` will return `key_2,opps,key_4`

Reply:
@soluchok Something is wrong, as per the current store implementation does nt seem like it is picking range its rather like of matching the end key with the start key. 

https://github.com/hyperledger/aries-framework-go/blob/d258db55ea379933cde5bacfd7030a91d19c446c/pkg/storage/couchdb/couchdbstore_test.go#L259

Reply:
@soluchok there is something missing, memstore is looking for prefix of the string to look through, where as couch db is also matching string . Are you sure that is the behavior we are looking in mysql.  

https://github.com/hyperledger/aries-framework-go/blob/d258db55ea379933cde5bacfd7030a91d19c446c/pkg/storage/mem/mem_store.go#L129

Reply:
As per the other implementation this works slightly different than this.

Reply:
In the above scenario response will be different as keys are stored or entered into the db in ascending order. So oops will be shifted to the end. There for the return value will be different  This works I have tested multiple scenarios. You can check the output here http://www.sqlfiddle.com/#!9/c13b02/10
SELECT * FROM test 
WHERE `key` >= 'abc_' AND `key` < 'mno~'

Reply:
I have added this and ran the exact same test other implementation are running for the iterator. They passed.